### ISSUE TITLE
Test & verify OS X compatibility.

### ISSUE THREAD

**Issue by evilsocket** [evilsocket/bettercap-ng#10]
The `arp.spoof` module doesn't seem to work on OSX, some testing would be needed, like compiling for OSX, launching the attack while capturing packets with Wireshark and see what's going on there.

**Comment by paolofuse** [comment]
Hi, if you want I can help you, just tell me instructions.

**Comment by evilsocket** [comment]
After a few tests and recent PR, the `arp.spoof` module itself seems to work, but packets are not correctly forwarded by the attacking computer therefore the network goes down, this implies further inspection and debugging of the `firewall/firewall_darwin.go` implementation.

**Comment by onura** [comment]
There is another issue with arp.spoof similar to #24 . This time Golang's net.ParseMac function cannot parse a Mac address if it is not 0 padded. If mac address of the target or gw needs 0 padding, arp.spoof does not work. 

I will look into the forwarding issue.

**Comment by evilsocket** [comment]
@onura just pushed a fix for that

**Comment by evilsocket** [comment]
Oh so it looks like the router I was testing the OSX build against is sh*t :D Waiting for more confirmations before closing this issue.

**Comment by evilsocket** [comment]
@droberson is it possible to have screenshots and (ideally) pcaps just to be 100% sure?

**Comment by droberson** [comment]
@evilsocket sure. Ill send one when I get home this evening.

**Comment by onura** [comment]
@evilsocket I tested it on macOS and it works well except the case fixed by #27 .

**Comment by evilsocket** [comment]
@droberson very appreciated

**Comment by evilsocket** [comment]
@onura you're the man! :D

**Comment by onura** [comment]
@evilsocket I couldn't see any call to EnableForwarding from other than http.proxy module. When I run arp.spoof without enabling http.proxy, the network goes down. Manually setting net.inet.ip.forwarding = 1 fixes it.

Can it be the same case which breaks your network? Should't EnableForwarding be called before starting arp spoof?

**Comment by evilsocket** [comment]
@onura 
![homer-simpson-doh](https://user-images.githubusercontent.com/86922/35473052-75fcb764-037a-11e8-81ad-d20d698f7cef.jpg)

**Comment by evilsocket** [comment]
@onura just committed a fix for that, could you please check now? thanks <3

**Comment by onura** [comment]
@evilsocket Thanks, the fix works perfect. However there is another thing which bothers me. Although I set a target for arp spoofing, lack of IP forwarding had broken all network. I have opened a PR #31 for it.

**Comment by evilsocket** [comment]
doh 2!

**Comment by evilsocket** [comment]
After other tests both me and @onura concluded that we can close this issue, OS X is natively supported now ^_^

**Comment by gregkeys** [comment]
this seems to be an issue again, for the life of me I can't get arp.spoof to work on osx

bettercap version 2.9
osx El Capitan 10.11.6

**Comment by evilsocket** [comment]
@gregkeys just tested 2.9 on High Sierra and it works like a charm, I'm afraid the problem is yours.
